src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx

```diff
@@ -101,9 +104,54 @@ const RuleEditor = createReactClass({
     Promise.all(promises).then(([environments, rule]) => {
       this.setState({environments, rule});
     });
-  },
+  }
+
+  fetchStatus() {
+    this.pollHandler();
+    this.startPolling();
+  }
+
+  pollHandler = async () => {
+    const {api, organization, project} = this.props;
+    const {uuid} = this.state;
+    const origRule = {...this.state.rule};
+
+    const {status, rule} = await api.requestPromise(
+      `/projects/${organization.slug}/${project.slug}/rule-async-tasks/${uuid}/`
+    );
+
+    if (status === 'pending') {
+      return;
+    }
+
+    this.stopPolling();
+    if (status === 'failed') {
+      // TODO(meredith): better error message - maybe get the error
+      // passed through from the endpoint?
+      addErrorMessage(t('An error occurred'));
+      this.setState({loading: false, rule: origRule});
```

Comment:
@MeredithAnya I am confused why we are setting the rule like this. Is it because the rule might have changed while the request is in-flight?

Reply:
also not sure why I had this, probs just need to add in the error